cMake/FreeCAD_Helpers/SetupSalomeSMESH.cmake

```diff
@@ -1,6 +1,10 @@
 macro(SetupSalomeSMESH)
 # -------------------------------- Salome SMESH --------------------------
 
+    # Allow using external SMESH
```

Comment:
CMake supports two search modes. If  you write `find_package(SMESH CONFIG)` then it's in config mode and either expects a file SMESHConfig.cmake or smesh-config.cmake.
If you write `find_package(SMESH REQUIRED)` then it's in module search mode and uses the FindSMESH.cmake file.

Because we already have the line `find_package(SMESH CONFIG)` that has worked quite well for many people the `find_package(SMESH REQUIRED)` should be used as fallback solution. Something like this:
```
find_package(SMESH CONFIG)
if (...) # successful
... # set some variables
else()
    find_package(SMESH REQUIRED)
    if (...) # successful
...
    else()
# raise an error
    endif()
endif()
```